packages/components/src/border-control/border-control-dropdown/component.tsx

```diff
@@ -199,7 +198,7 @@ const BorderControlDropdown = (
 					</HStack>
 				) : undefined }
 				<ColorPalette
-					className={ popoverContentClassName }
+					className="" // TypeScript is throwing error if not present.
```

Comment:
I'm probably missing something obvious but I couldn't see where the presence of `className` was being required given there is a `// @ts-nocheck` on the `ColorPalette` and it's not typed.

Reply:
I've removed this in favour of adding a default of `undefined` to the `ColorPalette`'s `className` prop ( https://github.com/WordPress/gutenberg/pull/40836/commits/8bd87370cd8dc1d6e61c785e691b7bdd3f528a9c ). 

My TypeScript knowledge is pretty limited however it appears that the `// @ts-nocheck` at the beginning of the `ColorPalette` file only skips the type checking there rather than skip generation of type declarations for it as well.

@ciampo is the approach I've taken here acceptable until the `ColorPalette` gets typed?

Reply:
You're right about this being an unexpected behavior — the "trick" that we usually employ here is to move the props destructuring from the function's signature, to the function's body

<details>
<summary>In this case:</summary>

```diff
diff --git a/packages/components/src/color-palette/index.js b/packages/components/src/color-palette/index.js
index f4130be032..f590383c5f 100644
--- a/packages/components/src/color-palette/index.js
+++ b/packages/components/src/color-palette/index.js
@@ -152,17 +152,19 @@ const extractColorNameFromCurrentValue = (
 	return __( 'Custom' );
 };
 
-export default function ColorPalette( {
-	clearable = true,
-	className = undefined,
-	colors,
-	disableCustomColors = false,
-	enableAlpha,
-	onChange,
-	value,
-	__experimentalHasMultipleOrigins = false,
-	__experimentalIsRenderedInSidebar = false,
-} ) {
+export default function ColorPalette( props ) {
+	const {
+		clearable = true,
+		className,
+		colors,
+		disableCustomColors = false,
+		enableAlpha,
+		onChange,
+		value,
+		__experimentalHasMultipleOrigins = false,
+		__experimentalIsRenderedInSidebar = false,
+	} = props;
+
 	const clearColor = useCallback( () => onChange( undefined ), [ onChange ] );
 	const showMultiplePalettes =
 		__experimentalHasMultipleOrigins && colors?.length;

```
</details>

It should be enough to keep TSC happy without introducing runtime changes :)

Reply:
Thanks, I've corrected the approach via https://github.com/WordPress/gutenberg/pull/40836/commits/cdf0cf183aa65deefdc99bf37e9345abad9a363a